### DISCUSSION TITLE
Want to use LaTeX-Workshop's pdf viewer as my default pdf viewer

### DISCUSSION THREAD

**Discussion by sunwoo-kim** [James-Yu/LaTeX-Workshop#4253] [Q&A] [answered]
Hello,

I was wondering if there was a way to use the latex-workshop's pdf viewer as the default pdf viewer for my mac. The reason is because I really like the dark mode on it, since I can use invert + hue rotate to make the colors the same as in white, and turn on based on use system settings, which I think is quite neat and I haven't seen before. 

I know there is `workbench.editorAssociations`, but it doesn't seem to be working. I set:
```
"workbench.editorAssociations": {
    "*.pdf": "latex-workshop-pdf-hook",
}
```

and I set vs code as the default programme to open pdfs. But it's somehow hit and miss - sometimes the pdf viewer opens when I open a file, but sometimes it loads the pdf as a text file. Other times it says it has bad text encoding.

I'm not sure why there is such an inconsistent behaviour. Is there a way to make it consistent?

Thanks in advance.

**Comment by James-Yu** [comment, marked as answer]
I would suggest you file an issue in vscode's repository on opening pdf as text file. This extension just provides a pdf hook to vscode, which determines when and how to call it.
